src/github.com/couchbase/sync_gateway/db/design_doc.go

```diff
@@ -117,27 +101,20 @@ func (db *Database) QueryDesignDoc(ddocName string, viewName string, options map
 
 // Cleans up the Value property, and removes rows that aren't visible to the current user
 func filterViewResult(input walrus.ViewResult, user auth.User) (result walrus.ViewResult) {
-	checkChannels := false
-	var visibleChannels ch.TimedSet
-	if user != nil {
-		visibleChannels = user.InheritedChannels()
-		checkChannels = !visibleChannels.Contains("*")
-	}
+
```

Comment:
Looks like the type assertion to string was extraneous, so I removed it.

> It looks like we're still calling filterViewResult if the admin is querying a non-internal doc. Is this needed?

It does appear to be needed for stripping _sync property, so I updated the function comments.

> I had thought we could just leave out the 'isInternalDDoc' condition altogether?

if it's an internal doc and the include_docs option was specified, then we have to strip
out the _sync field from the doc before returning.  (I added some comments to that effect)